File: sorting_alogs/Python/Bubble Sort/bubble_sorting.py
#Accept the number of array/list size in interger format
array_size =int(input("Enter the number of array size(Integer): "))

"""
Create a while loop that will that will work as validation to enter the valid array size.
Only accepting number greater than 1.

I have taken a long and complex for loop for this one
"""
#The below while loop 'in commet' is long of you want make some changes in it and test
#while (array_size==0) or (array_size<0) or (array_size==1) or (array_size.isdecimal) or (array_size.isdigit()) :
while array_size<2:
    array_size = int(input("Wrong entry! Array size cannot be 0, negative, or 1: "))

#Create an empty list to store the numbers
arrx = []
#Accept each number in the array and put it inside the arrx[] list
for x in range(array_size):
    num = int(input("Enter a number: "))
    arrx.append(num)
#Print the list
print("\nYour Array looks like this:\n",arrx)

#Lets jump to the main logic
#Create a counter with value 1 in while loop format
counter = 1;
while counter<array_size:
    print("\ncounter size after iteration is: ",counter,"\n")
    for i in range(array_size-counter):
        if arrx[i]>arrx[i+1]: #If the i elements is greater than the element i+1(ahead of it) then swap
            temp = arrx[i]
            arrx[i]=arrx[i+1]
            arrx[i+1]=temp
    #Print each iteration result till the sorting is complete.
    print("After the ",counter,"iteration, your array looks like this: ",arrx)
    #Increment the counter after each iteration so that once the counter!<array_size then final array will be printed
    counter+=1
print("Your sorted array looks like--> ",end=" ")
#Print the final array
for num in arrx:
    print(num,end=" ")

File: sorting_alogs/bubble_sorting.cpp
#include<iostream>

using namespace std;

int main() {
    cout<<"Enter the number array \n";
    int n;
    cin>>n; //Accept the size of the array
    //if you want to keep validation then use the while loop else skip it
    while (n<=0) {
      cout<<"Kindly enter the value of n greater than 0 \n";
      cin>>n;
    }
    //conver the number (array size) into proper array format
    int arr[n];
    //Hence now the array is created of n size

    //accept the values in for loop till the array is full 
    for (int i=0;i<n;i++) {
        cout<<"Enter the number:  ";
        cin>>arr[i];
    }
    //if you wan to print the array then use the below for loop
    cout<<"Your array looks like this:";
    for (int i=0;i<n;i++) {
        cout<<arr[i]<<" ";
    }

    //Now lets play with the main logic
  
    //declare a counter and compare with the array size so that the while loop will keep runnning with the entier logic
    int counter = 1;
    while(counter<n) //Declare the while loop till the counter the size of the array is satisfied hence (counter<n)
      { 
        //Declare for loop which keeps running inside the while loop
        for (int i=0;i<n-counter;i++) {
            if (arr[i]>arr[i+1])
            // the if condition checks if the vale in array is greater than +1 if yes then swip :)
             { 
                int temp = arr[i];
                arr[i] = arr[i+1];
                arr[i+1] = temp;
            }
        }
      //The below loop printes each sorted array
        cout<<"\nAfter each iteration "<<counter<<":";
        for (int i=0;i<n;i++) {
          cout<<arr[i]<<" ";
        }
        //keep incrementing the counter
        counter++;
    }
    //Declare a for loop which will print the out in sorted array
    cout<<"\n"<<"Your sorted array looks like this: ";
    for (int i=0;i<n;i++) {
        cout<<arr[i]<<" ";
    }
    cout<<endl;
    return 0; 
    //just return the code with +1 value looks cool :)
}

//end of the code
//Author @wizcoderx
